app/controllers/api/physical_storages_controller.rb

```diff
@@ -29,5 +29,11 @@ def delete_resource_action(type, id = nil, _data = nil)
         {:task_id => physical_storage.delete_physical_storage_queue(User.current_user)}
       end
     end
+
+    def validate_resource(type, _id = nil, data = {})
+      create_ems_resource(type, data, :supports => true) do |ems, klass|
```

Comment:
@TonySchneider I don't think `create_ems_resource` is what you're looking for here.  This is a convenience method for the create action where you're trying to validate.

You probably want to use `api_action` and then `ensure_supports(type, klass, :validate)`

Reply:
Yeah, I see since I actually don't need the full_action_results method to be called, and also the action should be different... got it.
Will handle it.